packages/server/api/routes/products.router.js

```diff
@@ -37,25 +37,29 @@ const productsController = require('../controllers/products.controller');
 router.get('/', (req, res, next) => {
   if (req.query.category) {
     productsController
-      .getProductsByCategory(req.query.category)
+      .getProductsByCategory(
+        req.query.category,
+        req.query.limit,
+        req.query.offset,
+      )
       .then((result) => res.json(result))
       .catch(next);
-  }
-  if (req.query.search) {
+  } else if (req.query.search) {
     productsController
       .getProductBySearch(req.query.search, res)
       .then((result) => res.json(result))
       .catch((error) => {
         next(error);
       });
-  }
-  try {
-    productsController
-      .getProducts()
-      .then((result) => res.json(result))
-      .catch(next);
-  } catch (error) {
-    res.status(404).json({ error: 'Bad Get Request' });
+  } else {
+    try {
```

Comment:
I'd wrap the whole router in this try/catch, any of those functions could throw.

Another thing to doubt about is why there are 3 different code paths here, what if I want to get things in category "shoes" but also search for "skateboarding"? 
Will you add a new if/else branch to check if I have both category and search in my query? 
I think it would be better if you figure out ways to support multiple options within the same request.

Reply:
Yeah ,but  search and allproducts is done by other team the category is done by our team the whole thing should be managed at some point